resources/examples/host/simpleResolver/test_simpleResolver.py

```diff
@@ -0,0 +1,131 @@
+#
+#   Copyright 2022 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+Test cases for the simpleResolver CLI.
+"""
+
+import json
+import subprocess
+import pathlib
+import pytest
+import sys
+
+
+from openassetio import BatchElementError
+from openassetio.hostApi import ManagerFactory
+
+# pylint: disable=no-self-use
+# pylint: disable=invalid-name,redefined-outer-name
+# pylint: disable=missing-class-docstring,missing-function-docstring
+
+
+class Test_simpleResolver_errors:
+    def test_when_config_env_not_set_then_message_printed_and_return_code_is_one(self):
+        result = execute_cli()
+        expected_message = [
+            "ERROR: No default manager configured, "
+            f"check ${ManagerFactory.kDefaultManagerConfigEnvVarName}"
+        ]
+        assert result.stderr.splitlines() == expected_message
+        assert result.returncode == 1
+
+    def test_when_config_set_and_plugin_path_not_set_then_error_and_return_code_wrapped(
+        self, test_config_env  # pylint: disable=unused-argument
+    ):
+        result = execute_cli()
+        expected_message = [
+            "\x1b[0;33m    warning: OPENASSETIO_PLUGIN_PATH is not set. "
+            "It is somewhat unlikely that you will find any plugins...\033[0m",
```

Comment:
This is something that already exists - rather than anything added by the example. It comes from the [PythonPluginSystemManagerImplementationFactory](https://github.com/OpenAssetIO/OpenAssetIO/blob/caabd2b95f126eced81486389de244704d4a25be/python/openassetio/pluginSystem/PythonPluginSystemManagerImplementationFactory.py#L71). There is a mild amount of whimsy in the project in general to attempt to keep people sane when they're bashing their head against a pipeline when things aren't working....

Reply:
Added #641 to make sure we're not being inadvertently vague in an attempt to be a "friendly" project.

Reply:
:+1: